cmd/kubeadm/app/cmd/reset_test.go

```diff
@@ -149,7 +149,6 @@ func TestConfigDirCleaner(t *testing.T) {
 		if err != nil {
 			t.Errorf("Unable to create temporary directory: %s", err)
 		}
-		defer os.RemoveAll(tmpDir)
```

Comment:
I'm not sure about this line change still. In case of failure in line 157, new line 186 will not be executed, and we will end-up with some tmpDir leftovers.

Reply:
@kad The failure of line 157 and 165 will cause some directory and file not created, but why it could cause new line 186 executed? If new line 186 is sure to execute, there is no problem with this commit. And there is another solution, we can move the content inside the outer loop to new function and call the function in the loop, the first *defer* can be reserved int this solution.

Reply:
I was thinking more along the line of something like this:
```go
topTempDir, err := ioutil.TempDir("", "kubeadm-reset-test-top")
if err != nil {
    t.Errorf("Unable to create temporary directory: %s", err)
}
defer os.RemoveAll(topTempDir)
for name, test := range tests {
   tmpDir, err := ioutil.TempDir(topTempDir, "kubeadm-reset-test")
   ...
}
```
This way this one `defer` will be out of loop, all test cases will be executed in subdirectories inside `topTempDir`, which will be cleaned after loop finishes. But ok, we can proceed with current variant as well.

Reply:
@kad It's OK with your solution above. But I think separating resources of each unit test will make code more safety and readable.